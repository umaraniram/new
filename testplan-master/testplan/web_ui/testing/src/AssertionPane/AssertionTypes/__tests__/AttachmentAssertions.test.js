/* Test the AttachmentAssertions components. */
import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import {
  AttachmentAssertion,
  MatplotAssertion
} from '../AttachmentAssertions.js';

const defaultAssertionProps = {
"category": "DEFAULT",
"machine_time": "2019-02-12T17:41:43.312797+00:00",
"description": null,
"line_no": 675,
"meta_type": "entry",
"utc_time": "2019-02-12T17:41:43.312789+00:00"
}

const defaultReportUid = "123";

describe('AttachmentAssertion', () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  it('renders an attached text file', () => {
    const assertionProps = {
      ...defaultAssertionProps,
      type: "Attachment",
      dst_path: 'tmpthpcdtwn-cd4f4c6e94971896a71b4a1d47785a90b19f6565-900.txt',
      filesize: 900,
      hash: 'cd4f4c6e94971896a71b4a1d47785a90b19f6565',
      orig_filename: 'tmpthpcdtwn.txt',
      source_path: '/tmp/tmpthpcdtwn.txt',
    }
    const shallowComponent = shallow(
      <AttachmentAssertion
        assertion={assertionProps}
        reportUid={defaultReportUid}
      />
    );
    expect(shallowComponent).toMatchSnapshot();
  });

  it('renders an attached image', () => {
    const assertionProps = {
      ...defaultAssertionProps,
      type: "Attachment",
      dst_path: 'tmpthpcdtwo-ad4f4c6e94971896a71b4a1d47785a90b19f6565-990.png',
      filesize: 990,
      hash: 'ad4f4c6e94971896a71b4a1d47785a90b19f6565',
      orig_filename: 'tmpthpcdtwo.png',
      source_path: '/tmp/tmpthpcdtwo.png',
    }
    const shallowComponent = shallow(
      <AttachmentAssertion
        assertion={assertionProps}
        reportUid={defaultReportUid}
      />
    );
    expect(shallowComponent).toMatchSnapshot();
  });

  it('renders an unknown filetype', () => {
    const assertionProps = {
      ...defaultAssertionProps,
      type: "Attachment",
      dst_path: 'tmpthpcdtwy-bd4f4c6e94971896a71b4a1d47785a90b19f6565-200.xyz',
      filesize: 200,
      hash: 'bd4f4c6e94971896a71b4a1d47785a90b19f6565',
      orig_filename: 'tmpthpcdtwy.xyz',
      source_path: '/tmp/tmpthpcdtwy.xyz',
    }
    const shallowComponent = shallow(
      <AttachmentAssertion
        assertion={assertionProps}
        reportUid={defaultReportUid}
      />
    );
    expect(shallowComponent).toMatchSnapshot();
  });

  it('renders an attachment using interactive API', () => {
    const assertionProps = {
      ...defaultAssertionProps,
      type: "Attachment",
      dst_path: 'tmpthpcdtwn-cd4f4c6e94971896a71b4a1d47785a90b19f6565-900.txt',
      filesize: 900,
      hash: 'cd4f4c6e94971896a71b4a1d47785a90b19f6565',
      orig_filename: 'tmpthpcdtwn.txt',
      source_path: '/tmp/tmpthpcdtwn.txt',
    }
    const shallowComponent = shallow(
      <AttachmentAssertion
        assertion={assertionProps}
        reportUid={null}
      />
    );
    expect(shallowComponent).toMatchSnapshot();
  });

});

describe('MatplotAssertion', () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  it('renders an image', () => {
    const assertionProps = {
      ...defaultAssertionProps,
      type: "Matplot",
      dst_path: 'tmpthpcdtwo-ad4f4c6e94971896a71b4a1d47785a90b19f6565-860.png',
      filesize: 860,
      hash: 'ad4f4c6e94971896a71b4a1d47785a90b19f6565',
      orig_filename: 'tmpthpcdtwo.png',
      source_path: '/tmp/tmpthpcdtwo.png',
    }
    const shallowComponent = shallow(
      <AttachmentAssertion
        assertion={assertionProps}
        reportUid={defaultReportUid}
      />
    );
    expect(shallowComponent).toMatchSnapshot();
  });
});
