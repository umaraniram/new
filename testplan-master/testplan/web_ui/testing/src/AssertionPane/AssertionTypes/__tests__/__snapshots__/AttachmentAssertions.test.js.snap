// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentAssertion renders an attached image 1`] = `
<Fragment>
  <Row
    tag="div"
  >
    <Col
      className="contentSpan_12mhtst"
      lg="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span>
        <div>
          <figure
            className="caption_775tb"
          >
            <img
              alt="Cannot Find File"
              className="img-responsive ..."
              src="/api/v1/reports/123/attachments/tmpthpcdtwo-ad4f4c6e94971896a71b4a1d47785a90b19f6565-990.png"
            />
            <figcaption
              className="caption_775tb"
            >
              <a
                href="/api/v1/reports/123/attachments/tmpthpcdtwo-ad4f4c6e94971896a71b4a1d47785a90b19f6565-990.png"
              >
                Image
              </a>
            </figcaption>
          </figure>
        </div>
      </span>
    </Col>
  </Row>
</Fragment>
`;

exports[`AttachmentAssertion renders an attached text file 1`] = `
<Fragment>
  <Row
    tag="div"
  >
    <Col
      className="contentSpan_12mhtst"
      lg="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span>
        <TextAttachment
          devMode={false}
          file_name="tmpthpcdtwn.txt"
          src="/api/v1/reports/123/attachments/tmpthpcdtwn-cd4f4c6e94971896a71b4a1d47785a90b19f6565-900.txt"
        />
      </span>
    </Col>
  </Row>
</Fragment>
`;

exports[`AttachmentAssertion renders an attachment using interactive API 1`] = `
<Fragment>
  <Row
    tag="div"
  >
    <Col
      className="contentSpan_12mhtst"
      lg="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span>
        <TextAttachment
          devMode={false}
          file_name="tmpthpcdtwn.txt"
          src="/api/v1/interactive/attachments/tmpthpcdtwn-cd4f4c6e94971896a71b4a1d47785a90b19f6565-900.txt"
        />
      </span>
    </Col>
  </Row>
</Fragment>
`;

exports[`AttachmentAssertion renders an unknown filetype 1`] = `
<Fragment>
  <Row
    tag="div"
  >
    <Col
      className="contentSpan_12mhtst"
      lg="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span>
        <a
          href="/api/v1/reports/123/attachments/tmpthpcdtwy-bd4f4c6e94971896a71b4a1d47785a90b19f6565-200.xyz"
        >
          tmpthpcdtwy.xyz
        </a>
      </span>
    </Col>
  </Row>
</Fragment>
`;

exports[`MatplotAssertion renders an image 1`] = `
<Fragment>
  <Row
    tag="div"
  >
    <Col
      className="contentSpan_12mhtst"
      lg="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <span>
        <div>
          <figure
            className="caption_775tb"
          >
            <img
              alt="Cannot Find File"
              className="img-responsive ..."
              src="/api/v1/reports/123/attachments/tmpthpcdtwo-ad4f4c6e94971896a71b4a1d47785a90b19f6565-860.png"
            />
            <figcaption
              className="caption_775tb"
            >
              <a
                href="/api/v1/reports/123/attachments/tmpthpcdtwo-ad4f4c6e94971896a71b4a1d47785a90b19f6565-860.png"
              >
                Image
              </a>
            </figcaption>
          </figure>
        </div>
      </span>
    </Col>
  </Row>
</Fragment>
`;
